src/sentry/api/endpoints/project_key_details.py

```diff
@@ -50,74 +91,74 @@ def put(self, request: Request, project, key_id) -> Response:
         serializer = ProjectKeyRequestSerializer(data=request.data, partial=True)
         default_version = get_default_sdk_version_for_project(project)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        result = serializer.validated_data
 
-            if result.get("name"):
-                key.label = result["name"]
-
-            if not key.data:
-                key.data = {}
+        if result.get("name"):
+            key.label = result["name"]
+        if not key.data:
+            key.data = {}
+        key.data["browserSdkVersion"] = (
+            default_version if not result.get("browserSdkVersion") else result["browserSdkVersion"]
+        )
 
-            key.data["browserSdkVersion"] = (
-                default_version
-                if not result.get("browserSdkVersion")
-                else result["browserSdkVersion"]
-            )
+        result_dynamic_sdk_options = result.get("dynamicSdkLoaderOptions")
+        if result_dynamic_sdk_options:
+            if key.data.get("dynamicSdkLoaderOptions"):
+                key.data["dynamicSdkLoaderOptions"].update(result_dynamic_sdk_options)
+            else:
+                key.data["dynamicSdkLoaderOptions"] = result_dynamic_sdk_options
+
+        if result.get("isActive") is True:
+            key.status = ProjectKeyStatus.ACTIVE
+        elif result.get("isActive") is False:
+            key.status = ProjectKeyStatus.INACTIVE
+
+        if features.has("projects:rate-limits", project):
+            ratelimit = result.get("rateLimit", -1)
+            if (
+                ratelimit is None
+                or ratelimit != -1
+                and ratelimit
+                and (ratelimit["count"] is None or ratelimit["window"] is None)
```

Comment:
I find this if statement and the elif below kind of confusing, is there any way we can refactor this (or maybe just add more brackets) to make whatever is happening here more clear?

Reply:
Maybe this is out of scope for this PR as well - not 100% sure

Reply:
A refactor like that is prob better in a separate PR. I agree that it's not great reading it tho (code is super old)

Reply:
Create a ticket for it? I'm not sure how we keep track of these things but it's a good idea and I agree it doesn't belong to this PR.